certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -124,11 +131,33 @@ def __init__(self, *args, **kwargs):
     @property
     def mod_ssl_conf_src(self):
         """Full absolute path to SSL configuration file source."""
-        config_filename = "options-ssl-nginx.conf"
-        if self.version < (1, 5, 9):
-            config_filename = "options-ssl-nginx-old.conf"
-        elif self.version < (1, 13, 0):
-            config_filename = "options-ssl-nginx-tls12-only.conf"
+        """
+        Why all this complexity? Well, we want to support Mozilla's intermediate
+        recommendations. But TLS1.3 is only supported by newer versions of Nginx.
+        And as for session tickets, our ideal is to turn them off across the board.
+        But! Turning them off at all is only supported with new enough versions of
+        Nginx. And older versions of OpenSSL have a bug that leads to browser errors
+        given certain configurations. While we'd prefer to have forward secrecy, we'd
+        rather fail open than error out. Unfortunately, Nginx can be compiled against
+        many versions of OpenSSL. So we have to check both for the two different features,
+        leading to four different combinations of options.
+        For a complete history, check out https://github.com/certbot/certbot/issues/7322
+        """
```

Comment:
smallest final thing, linter's unhappy that this is on its own line instead of a part of the function comment above